### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- Add more tests.
- Using the `assertIsResource` to check whether the expected type is `resource`.

### CODE REVIEW

File: tests/Commands/CommandRegistryTest.php [hunk 1]
```diff
@@ -134,6 +135,14 @@ public function testTryGettingHandlerReturnsFalseIfNoCommandWasFound(): void
         $this->assertNull($commandHandler);
     }
 
+    public function testTryGettingHandlerThrowsInvalidArgumentException(): void
```
Comment (davidbyoung) [inline tests/Commands/CommandRegistryTest.php]:
```suggestion
    public function testTryGettingHandlerOfInvalidTypeThrowsInvalidArgumentException(): void
```

File: tests/Output/StreamOutputTest.php [hunk 2]
```diff
@@ -54,6 +55,20 @@ public function testReadingLineReadsFromInputStream(): void
         $this->assertEquals('foo', $this->output->readLine());
     }
 
+    public function testReadingLineThrowsRuntimeException(): void
```
Comment (davidbyoung) [inline tests/Output/StreamOutputTest.php]:
```suggestion
    public function testReadingLineWhenNotAtEndOfFileThrowsRuntimeException(): void
```

File: tests/Output/StreamOutputTest.php [hunk 3]
```diff
@@ -54,6 +55,20 @@ public function testReadingLineReadsFromInputStream(): void
...
+    {
+        $inputStream = fopen(__DIR__ . '/../../composer.json', 'r');
```
Comment (davidbyoung) [inline tests/Output/StreamOutputTest.php]: For consistency with other tests that interact with files, can you add a `tmp` directory to the same directory as this unit test with a file inside called `output.txt`?  Then, add a couple lines of fake data to it, and use `__DIR__ . '/tmp/output.txt` as the `$inputStream` and `$outputStream`.

File: tests/Output/StreamOutputTest.php [hunk 4]
```diff
@@ -54,6 +55,20 @@ public function testReadingLineReadsFromInputStream(): void
...
+        $outputStream = fopen(__DIR__ . '/../../composer.json', 'r');
+        $output = new StreamOutput($outputStream, $inputStream, $this->compiler);
+        try {
+            $output->readLine();
+        } catch (RuntimeException $e) {
+            $this->assertSame('Failed to read line', $e->getMessage());
+            fclose($inputStream);
```
Comment (davidbyoung) [inline tests/Output/StreamOutputTest.php:71]: Put these two `fclose` into a `finally` block just in the case that `$output->readLine()` does not throw the expected `RuntimeException`.

**Review by davidbyoung — CHANGES_REQUESTED** [review]

**Comment by peter279k** [comment]
@davidbyoung, thank you for your reply.

I accept these request changes and push them in the latest commit.

Please check that :).

**Pull request merged**